core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+      expect(firstCollectionNode.getAcquiredSkillList().getSkills()).toEqual(
+          []);
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id0',
+          firstCollectionNode.getExplorationSummaryObject());
+      firstCollectionNode = collection.getCollectionNodeByExplorationId(
+          'exp_id0');
+      expect(
+          firstCollectionNode.getPrerequisiteSkillList().getSkills()).toEqual(
+              []);
+      expect(firstCollectionNode.getAcquiredSkillList().getSkills()).toEqual([
+          'exp title0']);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode]);
+    });
+
+    it('should correctly append a node to a non-empty collection', function() {
+      var collection = createLinearCollection();
+      var newCollectionNodeBackendObject = {
+        exploration_id: 'exp_id3',
+        prerequisite_skills: [],
+        acquired_skills: [],
+        exploration_summary: {
+          title: 'exp title3',
+          category: 'exp category',
+          objective: 'exp objective'
+        }
+      };
+      var newCollectionNode = CollectionNodeObjectFactory.create(
+          newCollectionNodeBackendObject);
+      expect(newCollectionNode.getPrerequisiteSkillList().getSkills()).toEqual(
+          []);
+      expect(newCollectionNode.getAcquiredSkillList().getSkills()).toEqual(
+          []);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id3',
+          newCollectionNode.getExplorationSummaryObject());
+      newCollectionNode = collection.getCollectionNodeByExplorationId(
+        'exp_id3');
+      expect(newCollectionNode.getPrerequisiteSkillList().getSkills()).toEqual(
+          ['exp title2']);
+      expect(newCollectionNode.getAcquiredSkillList().getSkills()).toEqual(
+          ['exp title3']);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([
+                collection.getCollectionNodeByExplorationId('exp_id0'),
+                collection.getCollectionNodeByExplorationId('exp_id1'),
+                collection.getCollectionNodeByExplorationId('exp_id2'),
+                collection.getCollectionNodeByExplorationId('exp_id3')]);
+    });
+  });
+
+  describe('shiftNodeLeft()', function() {
+    it('should handle shifting a node in a single node collection',
+        function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      firstCollectionNode.getAcquiredSkillList().setSkills(['exp title0']);
+      collection.addCollectionNode(firstCollectionNode);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
```

Comment:
Breaks after parameters should be two-space indented. Please update here and elsewhere. Overall, our JS style is to in general use 2 spaces for indentation, rather than 4.